index.bs

```diff
@@ -700,22 +721,56 @@ dictionary SensorOptions {
 
 A {{Sensor}} object has an associated [=sensor=].
 
+
+### Sensor task source ### {#task-source}
+
 Each {{Sensor}} object has a [=task source=]
 called a <dfn>sensor task source</dfn>, initially empty.
-A [=sensor task source=] can be enabled or disabled,
-and is initially enabled.
-When enabled,
+
+Issue(212): 
+
+A [=sensor task source=] can be enabled or disabled.
+
+To determine its initial state,
+run the steps to [=update the state of the sensor task source=].
+
+When the [=sensor task source=] is enabled,
 the [=event loop=] must use it as one of its [=task sources=].
 The [=task source=] for the [=tasks=] mentioned in this specification
 is the [=sensor task source=].
 
-When the [=visibility state=] of the [=Document=] in
+When the [=visibility state=] of the [=active document=] in
 the [=top-level browsing context=] changes,
-let |current_visibility_state| be the result of running
-the [=steps to determine the visibility state=] of the [=Document=].
-If |current_visibility_state| is "visible",
-enable the sensor task source,
-otherwise, disable it.
+the user agent must [=update the state of the sensor task source=].
+
+When an element that is the [=DOM anchor=] of a [=focusable area=] in any [=browsing context=] [=gains focus=]
+the user agent must [=update the state of the sensor task source=].
+ 
+<div algorithm>
+    
+    To <dfn export>update the state of the sensor task source</dfn>,
+    run the following steps:
+    
+    1.  Let |document| be the [=top-level browsing context=]'s [=active document=].
+    1.  Let |current_visibility_state| be the result of running
+        the [=steps to determine the visibility state=] of |document|.
+    1.  If |current_visibility_state| is not "visible", then:
+        1.  disable the [=sensor task source=].
+        1.  Return.
+    1.  If the [=currently focused area=] of the current [=top-level browsing context=]
+        is a [=nested browsing context=] whose [=active document=]'s [=origin=]
+        is not [=same origin-domain=] as |document|'s [=origin=], then:
+        1.  disable the [=sensor task source=].
+        1.  Return.
+    1.  If the [=currently focused area=] is in a different [=top-level browsing context=]
+        than the current [=top-level browsing context=]
+        and the [=origin=] of the [=active document=]
+        of that [=top-level browsing context=]
+        is not [=same origin-domain=] as |document|'s [=origin=], then:
+        1.  disable the [=sensor task source=].
```

Comment:
If you agree that task source is to be removed from the spec in favor of simple event dispatch, then there is no point to create an issue. If you think we should keep task source mechanism, then I would definitely create one.